File: cz/mg/vulkan/vk/PFNvkCmdPushDescriptorSetKHR.java
package cz.mg.vulkan.vk;

public class PFNvkCmdPushDescriptorSetKHR extends VkFunctionPointer {
    public PFNvkCmdPushDescriptorSetKHR() {
    }

    public PFNvkCmdPushDescriptorSetKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCmdPushDescriptorSetKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkCmdPushDescriptorSetKHR(long value) {
        setValue(value);
    }

    public PFNvkCmdPushDescriptorSetKHR(VkInstance instance) {
        super(instance, new VkString("vkCmdPushDescriptorSetKHR"));
    }

    public void call(VkCommandBuffer commandBuffer, VkPipelineBindPoint pipelineBindPoint, VkPipelineLayout layout, VkUInt32 set, VkUInt32 descriptorWriteCount, VkWriteDescriptorSet pDescriptorWrites){
        call(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.NULL_ADDRESS, pipelineBindPoint != null ? pipelineBindPoint.getVkAddress() : VkPointer.NULL_ADDRESS, layout != null ? layout.getVkAddress() : VkPointer.NULL_ADDRESS, set != null ? set.getVkAddress() : VkPointer.NULL_ADDRESS, descriptorWriteCount != null ? descriptorWriteCount.getVkAddress() : VkPointer.NULL_ADDRESS, pDescriptorWrites != null ? pDescriptorWrites.getVkAddress() : VkPointer.NULL);
    }

    private static native void call(long vkaddress, long commandBuffer, long pipelineBindPoint, long layout, long set, long descriptorWriteCount, long pDescriptorWrites);
}

File: c/cz/mg/vulkan/PFN_vkCmdPushDescriptorSetKHR.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkCmdPushDescriptorSetKHR_call(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong pipelineBindPoint, jlong layout, jlong set, jlong descriptorWriteCount, jlong pDescriptorWrites){
    (void)env;
    (void)jc;
    PFN_vkCmdPushDescriptorSetKHR f = (PFN_vkCmdPushDescriptorSetKHR)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkPipelineBindPoint*)jniLongToPointer(pipelineBindPoint)),
        *((VkPipelineLayout*)jniLongToPointer(layout)),
        *((uint32_t*)jniLongToPointer(set)),
        *((uint32_t*)jniLongToPointer(descriptorWriteCount)),
        ((VkWriteDescriptorSet*)jniLongToPointer(pDescriptorWrites))
    );
}
